### PR TITLE
feat: backend changes for customer searching

### PR DESCRIPTION
**Pull request by BrooksFlannery**
---
## Summary by cubic
Adds backend search for customers to support the table UI. Search by ID, name, or email, with a new endpoint to list available pricing model names.

- **New Features**
  - Cursor-paginated query supports exact ID search (trimmed) and partial name/email matches.
  - New route: listDistinctCustomerPricingModelNames returns deduplicated, case-insensitively sorted names for the organization.

<sup>Written for commit ec71ff5d28dc831e2da0c2d5e4d2364d3330ef04. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Search now supports exact customer-ID lookup in addition to name/email matching.

* **Improvements**
  * Customer search trimmed to handle whitespace and case-insensitive queries more robustly.
  * Public API surface cleaned up to remove unused/deprecated helpers.

* **Tests**
  * Added comprehensive tests covering ID, name, email searches, whitespace/case handling, org scoping, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/db/tableMethods/customerMethods.ts [hunk 1]
```diff
@@ -414,7 +435,62 @@ export const selectCustomersCursorPaginatedWithTableRowData =
...
+        .from(pricingModels)
+        .where(
+          and(
+            eq(pricingModels.id, customersTable.pricingModelId),
+            eq(pricingModels.name, pricingModelName)
+          )
+        )
+
+      return exists(pricingModelSubquery)
+    }
```
Comment (agreea) [inline platform/flowglad-next/src/db/tableMethods/customerMethods.ts]: Discussed offline: no need for this function anymore since we are only going to filter by pricing model id (rather than search by pm name)
